machineconfiguration/v1alpha1/types_internalreleaseimage.go

```diff
@@ -67,6 +67,19 @@ type InternalReleaseImageRef struct {
 
 // InternalReleaseImageStatus describes the current state of a InternalReleaseImage.
 type InternalReleaseImageStatus struct {
+	// conditions represent the observations of the InternalReleaseImage controller current state.
+	// Valid types are: Degraded.
+	// If Degraded is true, that means something has gone wrong in the controller.
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=20
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// observedGeneration represents the most recent generation of the InternalReleaseImage object observed by the controller.
+	// +kubebuilder:validation:XValidation:rule="self >= oldSelf", message="observedGeneration must not decrease"
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```

Comment:
Do you need a top-level observed generation?

Conditions can have an observed generation: https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition which is generally preferred over a top-level one